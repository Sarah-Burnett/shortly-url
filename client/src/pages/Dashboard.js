import React, { useState, Suspense, lazy } from 'react';
import styled from 'styled-components';
import { useLinks } from '../hooks/useLinks';
import { useProcessLink } from '../hooks/useProcessLink';
import { violet, grey } from '../components/Styles';
import DashNone from '../components/dash/DashNone';
import DashLoad from '../components/dash/DashLoad';
import DashLinks from '../components/links/DashLinks';
import Graph from '../components/dash/Graph';
const LinkCopier = lazy(() => import('../components/linkcrud/LinkCopier'));
const LinkEditor = lazy(() => import('../components/linkcrud/LinkEditor'));
const LinkDeletor = lazy(() => import('../components/linkcrud/LinkDeletor'));
const LinkShortener = lazy(() => import('../components/linkcrud/LinkShortener'));

const Dash = styled.div`
    min-height: 70vh;
    background: ${grey};
    color: ${violet};
    h2 {
        text-align: center;
    }
`

const Dashboard = () => {
    const { links } = useLinks();
    const { processLink: {process} } = useProcessLink();
    const [ loading, setLoading ] = useState(true);
    return (
        <Dash data-testid="dashboard">
            {loading && <DashLoad setLoading={setLoading}/> }
            {!loading && 
            <>
                {links.length > 0 ? <Graph/> : <DashNone/> }
            </>}
            <Suspense fallback={<div/>}>
                { !loading && links.length > 0 && <DashLinks/> }
                { process === "copy" && <LinkCopier/> }
                { process === "edit" && <LinkEditor/> }
                { process === "delete" && <LinkDeletor/> }
                <LinkShortener position="0"/>
            </Suspense>
        </Dash>
    )
};

export default Dashboard;