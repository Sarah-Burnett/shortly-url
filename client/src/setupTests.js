import React from 'react';
import ReactDOM from 'react-dom';
import '@testing-library/jest-dom/extend-expect';
import { render, cleanup, fireEvent, screen, waitForElement } from '@testing-library/react';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import renderer from 'react-test-renderer';
import { UserProvider } from './context/UserContext';
import { LinksProvider } from './context/LinksContext';
import { ActiveLinkProvider } from './context/ActiveLinkContext';
import { ProcessLinkProvider } from './context/ProcessLinkContext';
import { FlashProvider } from './context/FlashContext';
import App from './App';

export const ComponentWrapper = (Component) => {
    return (
        <UserProvider>
            <LinksProvider>
                <ActiveLinkProvider>
                    <ProcessLinkProvider>
                        <FlashProvider>
                            <Component/>
                        </FlashProvider>
                    </ProcessLinkProvider>
                </ActiveLinkProvider>
            </LinksProvider>
        </UserProvider>
    )
}

export const rendersWithoutCrashing = (Component) => {
    const div = document.createElement('div');
    ReactDOM.render(<Component/>, div);
}

export const matchesSnapshot = (Component) => {
    const tree = renderer.create(<Component/>).toJSON();
    expect(tree).toMatchSnapshot();
}

export const sampleLinks = [
    {
        "date":"2020-08-14T17:29:53.041Z",
        "click":[
            {"date":1597426302926},
            {"date":1597426306071},
            {"date":1597426310231},
            {"date":1597426314763},
            {"date":1597428206995},
            {"date":1597428210093},
            {"date":1597428215131},
            {"date":1597428218755}
        ],
        "_id":"K1pjMCK-F",
        "full":"http://reactcommunity.org/react-transition-group/css-transition",
        "userId":"5ebe7d235e4a501ee8699058",
        "short":"octopus","__v":8
    },
    {
        "date":"2020-08-14T17:33:55.748Z",
        "click":[],
        "_id":"lsncCe5RV",
        "full":"https://ba-grant.netlify.app/",
        "userId":"5ebe7d235e4a501ee8699058",
        "short":"bagrant","__v":0
    }
];
