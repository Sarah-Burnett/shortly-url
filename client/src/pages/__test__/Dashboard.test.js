import { rendersWithoutCrashing, matchesSnapshot, ComponentWrapper } from '../../setupTests';
import Dashboard from '../Dashboard';

const DashboardWrapper = () => ComponentWrapper(Dashboard);

it('renders dashboard without crashing', () => rendersWithoutCrashing(DashboardWrapper));

it('dashboard matches snapshot', () => matchesSnapshot(DashboardWrapper));

//TODO: test conditional rendering

