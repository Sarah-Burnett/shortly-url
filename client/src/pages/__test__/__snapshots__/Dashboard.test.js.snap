// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dashboard matches snapshot 1`] = `
<div
  className="sc-fzoLag fXtiGu"
  data-testid="dashboard"
>
  <div
    className="sc-AxiKw cKbaXL"
  >
    <p>
      Fetching Links...
    </p>
    <div
      className="sc-AxhCb cgHmUX"
    />
  </div>
  <div
    className="sc-fznWqX jhRSgX"
    style={
      Object {
        "top": "0",
      }
    }
  >
    <form
      className="sc-fzoXzr djigUv"
      onSubmit={[Function]}
    >
      <label
        htmlFor="link"
        style={
          Object {
            "display": "none",
          }
        }
      >
        Website Url
      </label>
      <p
        className="sc-fznJRM jLcXwS"
      />
      <input
        className="sc-fzpjYC kliCFF"
        data-testid="input"
        id="link"
        name="link"
        onChange={[Function]}
        onFocus={[Function]}
        required={true}
        value="Shorten a link here... "
      />
      <input
        className="sc-fzpjYC sc-fznxsB dSfCL"
        data-testid="button"
        disabled={false}
        type="submit"
        value="Shorten It!"
      />
    </form>
  </div>
</div>
`;
